src/FlowAware.cpp

```diff
@@ -211,6 +201,149 @@ void IR2Vec_FA::generateFlowAwareEncodings(std::ostream *o,
+  visitedFunctions.insert(function);
+  SmallVector<Function *, 15> funcStack;
+  funcStack.clear();
+  auto tmpParent = func2Vec(*function, funcStack);
+  // funcVecMap is updated with vectors returned by func2Vec
+  funcVecMap[function] = tmpParent;
+  auto calledFunctions = funcCallMap[function];
+  for (auto &calledFunction : calledFunctions) {
+    if (!calledFunction->isDeclaration() &&
+        visitedFunctions.count(calledFunction) == 0) {
+      // doing casting since calledFunctionsis of type of const llvm::Function*
+      // and we need llvm::Function* as argument
+      Function *foo = const_cast<Function *>(calledFunction);
+      // errs() << foo->getName().str() << "\n";
+      // This function is called recursively to update funcVecMap
+      updateFuncVecMap(foo, funcCallMap, visitedFunctions);
+    }
+  }
+}
+
+void IR2Vec_FA::generateFlowAwareEncodingsForFunction(
+    std::ostream *o, std::string name, std::ostream *missCount,
+    std::ostream *cyclicCount) {
+  // clock_t start = clock();
+  collectWriteDefsMap(M);
+  int noOfFunc = 0;
+  for (auto &f : M) {
+    auto funcName = f.getName().str();
+    // getting demangled function name
+    std::size_t sz = 17;
+    int status;
+    char *const readable_name =
+        __cxa_demangle(funcName.c_str(), 0, &sz, &status);
+
+    auto demangledName = status == 0 ? std::string(readable_name) : funcName;
+    // getting actual function name
+    size_t Size = 1;
+    char *Buf = static_cast<char *>(std::malloc(Size));
+    const char *mangled = funcName.c_str();
+    char *Result;
+    llvm::ItaniumPartialDemangler Mangler;
+    if (Mangler.partialDemangle(mangled)) {
+      Result = &demangledName[0];
+    } else {
+      Result = Mangler.getFunctionBaseName(Buf, &Size);
+    }
+    if (!f.isDeclaration() && Result == name) {
+      // If fname is matched with one of the functions in module, we will update
+      // funcVecMap of it and it's child functions recursively
+      std::unordered_set<const Function *> visitedFunctions;
+      updateFuncVecMap(&f, funcCallMap, visitedFunctions);
+    }
+  }
+  // iterating over all functions in module instead of funcVecMap to preserve
+  // order
+  for (auto &f : M) {
+    if (funcVecMap.find(&f) != funcVecMap.end()) {
+      auto calleelist = funcCallMap[&f];
+      Vector calleeVector(DIM, 0);
+      for (auto funcs : calleelist) {
+        // errs() << funcs->getName().str() << " ";
+        auto tmp = funcVecMap[funcs];
+        std::transform(tmp.begin(), tmp.end(), calleeVector.begin(),
+                       calleeVector.begin(), std::plus<double>());
+      }
+      // errs() << "\n";
+      scaleVector(calleeVector, WA);
+      auto tmpParent = funcVecMap[&f];
+      std::transform(calleeVector.begin(), calleeVector.end(),
+                     tmpParent.begin(), tmpParent.begin(), std::plus<double>());
+      funcVecMap[&f] = tmpParent;
+    }
+  }
+
+  for (auto &f : M) {
```

Comment:
This logic is very similar to what we have in `generateFlowAwareEncodings()`. Please try creating a function and use it in both the places.